src/Application/Services/ItemService.cs

```diff
@@ -39,86 +46,97 @@ public async Task<Item> Get(string key)
     {
         ItemEntity itemEntity = await _itemRepository.Get(key) ?? throw new NotFoundException("Key not found");
 
-        Item item = new()
+        if (itemEntity.ExpirationDate > DateTime.UtcNow)
         {
-            Key = itemEntity.Key,
-            Value = JsonSerializer.Deserialize<List<object>>(itemEntity.Value),
-            ExpirationPeriod = itemEntity.ExpirationPeriod,
-            ExpirationDate = itemEntity.ExpirationDate
-        };
+            DateTime updatedDate = DateTime.UtcNow.AddSeconds(itemEntity.ExpirationPeriod);
 
-        return item;
-    }
-
-    public async Task<string> Create(ItemCreate itemDto)
-    {
-        if (await _itemRepository.Get(itemDto.Key) is null)
-        {
-            int ExPeriod = itemDto.ExpirationPeriod ?? 0;
-
-            ItemEntity itemEntity = new()
+            Item item = new()
             {
-                Key = itemDto.Key,
-                Value = JsonSerializer.Serialize(itemDto.Value),
-                ExpirationPeriod = ExPeriod,
-                ExpirationDate = DateTime.Now.AddSeconds(ExPeriod)
+                Key = itemEntity.Key,
+                Value = JsonSerializer.Deserialize<List<string>>(itemEntity.Value),
+                ExpirationPeriod = itemEntity.ExpirationPeriod,
+                ExpirationDate = updatedDate
             };
 
-            await _itemRepository.Create(itemEntity);
+            await _itemRepository.UpdateExDate(itemEntity.Key, updatedDate);
 
-            return "Item created";
+            return item;
         }
         else
         {
-            int ExPeriod = itemDto.ExpirationPeriod ?? 0;
+            await _itemRepository.Delete(key);
 
-            ItemEntity itemEntity = new()
+            throw new NotFoundException();
+        }
+
+    }
+
+    public async Task<string> Create(ItemCreate itemDto)
+    {
+        if (itemDto.ExpirationPeriod < MaxItemExPeriod)
+        {
+            if (await _itemRepository.Get(itemDto.Key) is null)
             {
-                Key = itemDto.Key,
-                Value = JsonSerializer.Serialize(itemDto.Value),
-                ExpirationPeriod = ExPeriod,
-                ExpirationDate = DateTime.Now.AddSeconds(ExPeriod)
-            };
+                int exPeriod = itemDto.ExpirationPeriod is 0 ? DefaultExPeriod : itemDto.ExpirationPeriod.Value;
```

Comment:
incorrect code, should be "If expiration period is not defined - default value should be used from config"